lib/FundDistribution/FundDistributionFields/FundDistributionFieldsFinalContainer.js

```diff
@@ -36,26 +39,51 @@ const FundDistributionFieldsFinalContainer = ({
   const [expenseClassesByFundId, setExpenseClassesByFundId] = useState({});
 
   useEffect(() => {
-    const expenseClassesFundsToFetch = fundDistribution?.filter(({ fundId }) => (
-      fundId && !expenseClassesByFundId[fundId]
-    ));
-
-    if (expenseClassesFundsToFetch?.length) {
-      expenseClassesFundsToFetch.forEach(({ fundId }) => fetchExpenseClasses(
-        fundId, expenseClassesByFundId, setExpenseClassesByFundId, mutator.fundExpenseClasses,
-      ));
+    const uniqueFundDistribution = uniqBy(fundDistribution, 'fundId');
+
+    uniqueFundDistribution.forEach(({ fundId }) => fetchExpenseClasses({
+      fundId,
+      fiscalYearId,
+      setExpenseClassesByFundId,
+      mutator: mutator.fundExpenseClasses,
+    }));
+  }, [fundDistribution, fiscalYearId, mutator.fundExpenseClasses]);
+
+  const resetExpenseClasses = useCallback(() => {
+    formValues?.adjustments?.forEach(({ fundDistributions }, index) => {
+      // reset form field state if fundId is changed and there is no expense class for this fund
+      fundDistributions?.forEach(({ fundId, expenseClassId }, distributionIndex) => {
+        const fieldName = `adjustments[${index}].fundDistributions[${distributionIndex}]`;
```

Comment:
this exists only for invoices, you introduce potential issues for orders, please keep invoice related logic in invoices, it's not common logic

to be honest I don't understand what you try to reset, previously it worked somehow without this logic

Reply:
If you want I will remove the logic but it may create an issue. Let's discuss this tomorrow together.